src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -624,6 +618,16 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
 
   let openFilesA = openFiles |> AMap.map' (fun v -> v :> aval<_>)
 
+  let cancelAllOpenFileCheckRequests () =
+    transact (fun () ->
+      let files = openFiles |> AMap.force
+
+      for (_, fileVal) in files do
+        let (oldFile, cts: CancellationTokenSource) = fileVal |> AVal.force
+        cts.Cancel()
+        cts.Dispose()
+        fileVal.Value <- oldFile, new CancellationTokenSource())
+
```

Comment:
Calls cancel on all openfiles's `CancellationTokenSource` to stop type checking.